api/src/main/java/uk/gov/hmcts/payment/api/reports/FeesService.java

```diff
@@ -65,6 +66,7 @@ public FeesService(FeesRegisterClient feesRegisterClient) {
         return mapOfFeeVersionsDtoMap;
     }
 
+    @Cacheable("map")
```

Comment:
You try to cache a method result, which could return an empty map here which could cause some weird behavior. If the first return of the method is an empty map then the result remains empty until the cache is evicted. It returns a member variable which is a kind of caching as I see. Either the built in cache, or the custom solution should be used

Reply:
I guess, spring handles that with an expression like this  ? `@Cacheble("value = "feeCache", unless = "#result==null or #result.size()==0`)